cmd/client.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/thegreenwebfoundation/grid-intensity-go/pkg/provider"
+)
+
+func getClient(providerName string, cacheFile string) (provider.Interface, error) {
+	var client provider.Interface
+	var err error
+
+	switch providerName {
+	case provider.CarbonIntensityOrgUK:
+		c := provider.CarbonIntensityUKConfig{}
+		client, err = provider.NewCarbonIntensityUK(c)
+		if err != nil {
+			return nil, fmt.Errorf("could not make carbon intensity uk provider, %w", err)
+		}
+	case provider.ElectricityMap:
+		token := os.Getenv(electricityMapAPITokenEnvVar)
+		if token == "" {
+			return nil, fmt.Errorf("%q env var must be set", electricityMapAPITokenEnvVar)
+		}
+
+		c := provider.ElectricityMapConfig{
+			Token: token,
+		}
+		client, err = provider.NewElectricityMap(c)
+		if err != nil {
+			return nil, fmt.Errorf("could not make electricity map provider, %w", err)
+		}
+	case provider.Ember:
+		client, err = provider.NewEmber()
```

Comment:
Hmm I'm not sure on this. I thought we wanted to keep the ElectricityMap provider?

For Ember right now we're using the embedded data but we could extend this provider to fetch the monthly data instead?

Or if we wanted to support both the data file and fetching data I think we'd need to add another arg to `grid-intensity` to specify a time period. WDYT?

Reply:
oh, sorry - I should have been clear - I think we'd still totally keep the electricitymap provider. I was referring to `Ember/NewEmber`, and should have been more specific.

Yeah, I'm in favour of being able to extend the provider with extra command line args rather than having a separate provider for this. I'll create an issue to discuss that separately though.

Reply:
Ah got it thanks for clarifying.

Go doesn't strictly have constructors but this is the equivalent.
https://go.dev/doc/effective_go#composite_literals

I agree better to add optional parameters in the CLI instead of adding another provider.